src/dsp_tools/commands/xmllib/models/dsp_base_resources.py

```diff
@@ -0,0 +1,269 @@
+from __future__ import annotations
+
+import warnings
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Any
+
+from lxml import etree
+
+from dsp_tools.commands.xmllib.models.values import ColorValue
+from dsp_tools.commands.xmllib.models.values import LinkValue
+from dsp_tools.commands.xmllib.models.values import SimpleText
+from dsp_tools.commands.xmllib.value_checkers import find_geometry_problem
+from dsp_tools.commands.xmllib.value_checkers import is_decimal
+from dsp_tools.commands.xmllib.value_checkers import is_integer
+from dsp_tools.commands.xmllib.value_checkers import is_string
+from dsp_tools.models.custom_warnings import DspToolsUserWarning
+
+XML_NAMESPACE_MAP = {None: "https://dasch.swiss/schema", "xsi": "http://www.w3.org/2001/XMLSchema-instance"}
+DASCH_SCHEMA = "{https://dasch.swiss/schema}"
+
+
+@dataclass
+class AnnotationResource:
+    res_id: str
+    label: str
+    comments: list[str]
+    annotation_of: str
+    permissions: str = "res-default"
+
+    def __post_init__(self) -> None:
+        _check_and_warn_strings(self.res_id, self.res_id, "Resource ID")
+        _check_and_warn_strings(self.res_id, self.label, "Label")
+
+    def serialise(self) -> etree._Element:
```

Comment:
how about `to_xml()`? It's similarly not quite correct, but maybe it's closer to correct?

Reply:
The reason why I chose it, is because if we ever change the output format, we would not need to rename the function. The way we structured it, that would not constitute a breaking change with regard to usage...

Reply:
It's a good consideration, but it's very unlikely that we will stop using XML all-together - and if so, that would break this lib anyways, because it should no longer be called `xmllib`... But I don't feel strongly about this change

Reply:
That is true.